[coq] Deprecate Coq Lang < 0.8 (#7599)
ocaml/dune: CHANGES.md

- On nix+macos, pass `-f` to the codesign hook to avoid errors when the binary
  is already signed (#7183, fixes #6265, @greedy)

> - Coq build languages less than 0.8 are deprecated, and will be

Reviewer: ```suggestion
- Coq build languages less than version 0.8 are deprecated, and will be
```
Would it be beneficial to include "version" in here (if it's accurate!)?

  removed in the next Dune version. All users are required to migrate
  to `(coq lang 0.8)` which provides the right semantics for theories

>   that have been globally installed, such as those coming from OPAM

Reviewer: ```suggestion
  that have been globally installed, such as those coming from opam
```

- Fix RPC server on Windows (used for OCaml-LSP). (#7666, @nojb)

> - Coq build languages less than version 0.8 are deprecated, and will

Reviewer 2: ```suggestion
- Coq lang versions less than version 0.8 are deprecated, and will
```
